scripts/pi-hole/php/groups.php

```diff
@@ -58,7 +58,10 @@ function JSON_error($message = null)
 } elseif ($_POST['action'] == 'add_group') {
     // Add new group
     try {
-        $names = str_getcsv(trim($_POST['name']), ' ');
+        // Payload from the web interface will have been sanitised with " converted to &quot;
+        // so before we process the input, we will do the opposite and replace &quot; with ";
+        $input = str_replace("&quot;","\"",trim($_POST['name']));
```

Comment:
Just to confirm, shouldn't it make sense to use something like https://www.php.net/manual/en/function.htmlspecialchars-decode.php?

Reply:
Fair shout, I didn't know of it's existence before now. Either or works in this case.